File: java-context-os/src/main/java/com/owencli/contextos/pipeline/middleware/LLMMiddleware.java
package com.owencli.contextos.pipeline.middleware;

import com.owencli.contextos.core.base.BaseLLMClient;
import com.owencli.contextos.pipeline.PipelineContext;
import com.owencli.contextos.pipeline.PipelineMiddleware;
import java.util.concurrent.CompletableFuture;

public class LLMMiddleware implements PipelineMiddleware {
    private final BaseLLMClient c;
    public LLMMiddleware(BaseLLMClient cl) { this.c = cl; }
    public String name() { return "llm"; }
    public int order() { return 600; }
    public CompletableFuture<Void> execute(PipelineContext ctx) {
        return c.complete(ctx.packagedContext().getRawPrompt())
            .thenAccept(r -> ctx.setLlmResponse(String.valueOf(r)));
    }
}

File: context_os/pipeline/middlewares/llm_middleware.py
"""LLMMiddleware — LLM 调用。"""

from context_os.llm.client import BaseLLMClient
from context_os.pipeline.context import PipelineContext
from context_os.pipeline.middleware import PipelineMiddleware


class LLMMiddleware(PipelineMiddleware):
    """调用 LLM API。"""

    def __init__(self, llm_client: BaseLLMClient):
        self._c = llm_client

    def name(self) -> str:
        return "llm"

    def order(self) -> int:
        return 600

    async def execute(self, ctx: PipelineContext) -> None:
        response = await self._c.complete(ctx.packaged_context.raw_prompt)
        ctx.llm_response = str(response)
